rest_framework/serializers.py

```diff
@@ -997,13 +997,19 @@ def update(self, instance, validated_data):
         # relationships as being a special case. During updates we already
         # have an instance pk for the relationships to be associated with.
         m2m_fields = []
+        partial_update_extra_fields = self.get_partial_update_extra_fields()
+        update_fields = [*(partial_update_extra_fields or [])]
```

Comment:
It would be much simplier here to have only one variable and to have `.get_partial_update_extra_fields` to always return a list (or a tuple).

```suggestion
        update_fields = self.get_partial_update_extra_fields()
```